File: Backend/nuka-pos/src/main/java/com/nuka/nuka_pos/api/product/Product.java
package com.nuka.nuka_pos.api.product;

import com.nuka.nuka_pos.api.category.Category;
import com.nuka.nuka_pos.api.organization.Organization;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
@Table(name = "products")
@NoArgsConstructor
@AllArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "organization_id")
    private Organization organization;

    @ManyToOne(optional = false)
    @JoinColumn(name = "category_id")
    private Category category;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal cost;

    @Column(length = 100)
    private String barcode;

    @Column(nullable = false)
    private Boolean isActive = true;
}

File: Frontend/Nuka_POS/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String description;
  final double price;
  final double cost;
  final String? barcode;
  final bool isActive;
  final int organizationId;
  final int categoryId;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.cost,
    this.barcode,  // Nullable barcode field
    required this.isActive,
    required this.organizationId,
    required this.categoryId,
  });

  // Convert Product to JSON
  Map<String, dynamic> toJson() {
    final data = {
      'name': name,
      'description': description,
      'price': price,
      'cost': cost,
      'barcode': barcode,  // Include barcode only if not null
      'isActive': isActive,
      'organization': {'id':organizationId},
      'category': {'id':categoryId},
    };

    return data;
  }

  // Create a Product from JSON response
  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'],
      cost: json['cost'],
      barcode: json['barcode'],  // Handle nullable barcode
      isActive: json['isActive'],
      organizationId: json['organizationId'],
      categoryId: json['categoryId'],
    );
  }
}
